### DISCUSSION TITLE
Verify password via API client

### DISCUSSION THREAD

**Discussion by danielpox** [supabase/supabase#8296] [Feature Requests]
I'm in a situation where I need the user to verify their password in order to proceed.

**Note:** This is not related to signup or email verification. But rather for the logged in user to verify their password in order to unlock certain actions.

Currently I can't find any way to achieve this behavior via `supabase-js`.

My initial thought was to use the `auth.signIn()`, but this refreshes the access token and invalidates the application state.

What I'm looking for is a simple API method to verify the password for the currently signed in user (unless this introduces some security concerns).
Something like this:
`supabase.auth.api.verifyPassword('a-password')`. (Maybe not the `.api` scope, but just `supabase.auth`?)

In Firebase there's something similar called `reauthenticateWithCredential()`, and although this also updates the token, it didn't work to just swap Firebase with Supabase for my project.

My application is using the access token as an API caching parameter (via react-query), so whenever the token changes it re-fetches the data from my backend. Currently via the `auth.signIn()` method I'm getting some poor UX and user/token mismatch errors.

Unless this feature request would introduce a security concern (perhaps brute force), it would be a very handy feature to unlock parts of applications or enable client-side encryption.

Thanks in advance!

**Comment by GaryAustin1** [comment]
There is probably enough info in this thread to show you how to use an rpc/postgres function to take a user password for the logged in user and compare it to the one stored in auth.users() by encrypting it and comparing the two encrypted results.   The code shown in this is more complex in that it is also allowing to change the password.
https://github.com/supabase/supabase/discussions/4042

Edit:  One thing to note though, if you don't do what ever operations you need the password verified for in the rpc/postgres function itself, a user can fake the result of the function, just like any other operation done on the client side.

**Reply by danielpox** [reply]
Thanks @GaryAustin1! I'll try it out to see if it works for my use case.

> Edit: One thing to note though, if you don't do what ever operations you need the password verified for in the rpc/postgres function itself, a user can fake the result of the function, just like any other operation done on the client side.

I'm using the user's password to encrypt data client-side. So it's just to verify that the user entered the correct value before encrypting/decrypting.
